### PR TITLE
fix dirs

### PR DESCRIPTION
**Pull request by w-e-w**
### fixed issues

lora dir can be change by user to any dir using --lora-dir
or --data-dir when --lora-dir not in use

tldr: use `shared.cmd_opts.lora_dir` to get the dir and not a relative path you're using

https://github.com/AUTOMATIC1111/stable-diffusion-webui/wiki/Command-Line-Arguments-and-Settings

> it can also be in shared.cmd_opts.lyco_dir_backcompat 
but it didn't bother implemented in it

---

fix know_dir 
extension installation location can also change

### other improvements that you might want to implement

well this is not crucial generally speaking you should use the funcrions in webui/module/hashes.py to calculate the hashes of files, reason being that the results can be cached and reused

---

about the lora list drowdown choices
currently you scan the lora as webui launch the results will get saved in to gradio and reuse every time when the page loads or reloads, later on the page the user can manually refresh the list

what you may not have realized is that but you may not have realized is that the list that is scanned at the initial loading will not be refresh when the mentally refreshes
which means if you add som new lora after webui launches, manually rescan rescan (new lora appears on the list), refresh the webpage (new lora is gone)

one way to fix this is to rescan lora on every page load
you can do so by add add something like this, this will run on every page load
```py
    def ui(self, is_img2img):
        with gr.Blocks(analytics_enabled=False) as block:
                block.load(reload_lora_list, outputs=[lora_dropdown])
# this should be close enough but I did not test it
```

### CODE REVIEW

**Comment by Avaray** [comment]
Thank you 👍 

I think this can be related with https://github.com/Avaray/lora-keywords-finder/issues/11
